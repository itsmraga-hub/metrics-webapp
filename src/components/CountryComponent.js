import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

import style from './css/CountryComponent.module.css';

const CountryComponent = () => {
  const [statePending, updateState] = useState(false);
  const [All, updateAll] = useState({});
  const state = useSelector((state) => state.cases);
  const { filterCountry, pending } = state;
  useEffect(() => {
    if (!pending) {
      updateAll(filterCountry[0].All);
      updateState(true);
    }
  }, [pending, All]);

  return (
    <>
      {
      statePending ? (
        <div className={style.Container}>
          <Link to={{ pathname: '/' }}>
            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
            <button type="button"><i className="fa-solid fa-circle-chevron-left fa-2x" /></button>
          </Link>
          <h1>Country statistics</h1>
          <h2>
            {All.country}
            {' '}
            <span>
              (
              {All.abbreviation}
              )
            </span>
          </h2>
          <div>
            <p>
              Continent:
              {' '}
              <span>{All.continent}</span>
            </p>
            <p>
              Capital city:
              {' '}
              <span>{All.capital_city}</span>
            </p>
            <p>
              Location:
              {' '}
              <span>{All.location}</span>
            </p>
            <p>
              Area covered:
              {' '}
              <span>{All.sq_km_area}</span>
            </p>
            <p>
              Population:
              {' '}
              <span>{All.population}</span>
            </p>
            <p>
              Confirmed cases:
              {' '}
              <span>{All.confirmed}</span>
            </p>
            <p>
              Deaths:
              {' '}
              <span>{All.deaths}</span>
            </p>
            <p>
              Last updated:
              {' '}
              <span>{All.updated}</span>
            </p>
          </div>
        </div>
      ) : (
        <div className={style.AwaitContainer}>
          <h1>Fetching data...</h1>
        </div>
      )
      }
    </>
  );
};

export default CountryComponent;
