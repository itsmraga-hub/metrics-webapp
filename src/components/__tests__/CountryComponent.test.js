import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configureStore';

import CountryComponent from '../CountryComponent';

it('<CountryComponent /> Renders Properly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <CountryComponent />
      </Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
