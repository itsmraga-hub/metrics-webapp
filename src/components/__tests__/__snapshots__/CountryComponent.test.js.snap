// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryComponent /> Renders Properly 1`] = `
<div
  className="AwaitContainer"
>
  <h1>
    Fetching data...
  </h1>
</div>
`;
